Ball.java

```diff
@@ -25,12 +21,19 @@ public Ball(String name, int x, int y)
 
 	@Override
 	public void update() {
+		mandatory_update();
 		bounce();
-		x_pre = x;y_pre = y;
-		x+=hspeed;y+=vspeed;
+		
+		// might move this to mandatory_update???
```

Comment:
I'm trying to make it so working with this is as easy as possible. If I move the variables vspeed and hspeed into GameObject every game object can use them. There is only one way to use them and that is this line. It is just a design question, doesn't really matter either way.